### PR TITLE
auto generate gha auth token

### LINKED ISSUE #4822: `metarelate.net`
## 📰 Custom Issue

The link check is failing against `metarelate.net`, which is linked from the 3.2 whatsnew

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR enables automated authenticating with a newly created and installed SciTools owned GitHub App for generating tokens for a GHA to enable other workflow actions.

The GitHub App `iris-actions` is only installed and enabled for `iris`.

The context here is to allow the `refresh-lockfiles` GHA to trigger the CI workflows automatically on the PR generated to update the `conda-lock` files on the `main` branch.

Reference: `create-pull-request` GHA [Authenticating with GitHub App generated tokens](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#authenticating-with-github-app-generated-tokens)

Closes #4822 

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by jamesp** [comment]
Nice clever workaround.

I assume this needs additional infrastructure in the form of the app that provides `secrets.AUTH_APP_ID` etc? That needs maintainer documentation somewhere, either in the docs, or in a comment in this file would probably be a good place.

**Comment by bjlittle** [comment]
I guess I could add something to the dev docs, that if anything point people to the standard GH docs etc on how to do all this.

We didn't have to go off piste from the minimal GH app to make this work, so it's pretty straight forward thankfully.

Also, I tested the same workflow on `geovista`, which was experiencing the same problem, and it works a treat. See https://github.com/bjlittle/geovista/pull/79 as an example with the CI workflows are successfully run automatically on the GHA generated PR.

**Comment by bjlittle** [comment]
@jamesp Supporting owner/developer/admin documentation is available [here](https://bjlittle-iris.readthedocs.io/en/latest/developers_guide/github_app.html).

File: docs/src/conf.py
Comment (trexfeathers) [inline docs/src/conf.py:361]: Some of us have discussed removing this link entirely (it's just in a What's New), since the website is unlikely to come back.

**Comment by jamesp** [comment]
> @jamesp Supporting owner/developer/admin documentation is available [here](https://bjlittle-iris.readthedocs.io/en/latest/developers_guide/github_app.html), as part of this PR.
> 
> On first impression the level of documentation may seem a tad overkill, but I suspect the process of creating a GitHub App is very rare, so it's probably best to capture the detail now before it fades 👍

Blimey, love it... I was only expecting a comment in the GHA file! :D

File: docs/src/conf.py
Comment (rcomer) [reply docs/src/conf.py]: We don't _have_ to hold up PRs with this sort of thing 😉 #4140

**Review by trexfeathers — COMMENTED** [review]
This looks good to me, although I'd like to get the go-ahead from @jamesp, too.

@bjlittle given the number of images used in the docs page, would you consider popping them all into their own sub-directory?

**Comment by jamesp** [comment]
I'm happy @trexfeathers

**Comment by bjlittle** [comment]
> @bjlittle given the number of images used in the docs page, would you consider popping them all into their own sub-directory?

@trexfeathers I'm confused... they are all in their own sub-directory i.e. `docs/src/developers_guide/assets`... or do you mean something else?

**Comment by trexfeathers** [comment]
> > @bjlittle given the number of images used in the docs page, would you consider popping them all into their own sub-directory?
> 
> @trexfeathers I'm confused... they are all in their own sub-directory i.e. `docs/src/developers_guide/assets`... or do you mean something else?

A sub-directory specifically for this page, since I personally find it difficult to work with documentation when there is a large mass of image assets that could belong to one of several pages.

`docs/src/developers_guide/assets/github_app` or similar.

**Comment by trexfeathers** [comment]
Jeez, I hadn't realised it was already a bit of a mess ☹ I don't like either of the apparent choices:

1. Confusing
   OR
2. Over-complicated

I'm just going to merge it

**Review by trexfeathers — APPROVED** [review]

**Pull request merged**
